feeds/wlan-ap/opensync/src/src/sm/src/ubus_collector.c

```diff
@@ -0,0 +1,668 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+
+#include "ubus_collector.h"
+
+/* Global list of events received from hostapd */
+dpp_event_report_data_t g_report_data;
+
+/* Internal list of processed events ready to be deleted from hostapd */
+static dpp_event_report_data_t *deletion_pending = NULL;
+static struct ubus_context *ubus = NULL;
+static char *ubus_object = NULL;
+
+typedef struct {
+	uint64_t session_id;
+	char *bss;
+
+	ds_dlist_node_t node;
+} delete_entry_t;
+
+static int client_first_data_event_cb(struct blob_attr *msg,
+				      dpp_event_record_session_t *dpp_session,
+				      uint64_t event_session_id)
+{
+	int error = 0;
+	struct blob_attr *tb_client_first_data_event[__CLIENT_FIRST_DATA_MAX] = {
+		0
+	};
+	char *mac_address = NULL;
+	uint64_t session_id = event_session_id;
+	uint32_t timestamp = 0;
+	uint64_t rx_ts = 0;
+	uint64_t tx_ts = 0;
+	dpp_event_record_first_data_t *first_data_event_dpp = NULL;
+
+	error = blobmsg_parse(client_first_data_event_policy,
+			      __CLIENT_FIRST_DATA_MAX,
+			      tb_client_first_data_event, blobmsg_data(msg),
+			      blobmsg_data_len(msg));
+	if (error)
+		return -1;
+
+	for (int i = 0; i < __CLIENT_FIRST_DATA_MAX; i++) {
```

Comment:
did the compiler not complain about this one ? the var declaration should go to the start of the function

Reply:
It's a counter var local to the loop. Why would the compiler complain?

Reply:
because C99 says that one should not define the counter inside the for block. nothing I really care about, opensync probably is very slack with its compiler flags. --pedantic certainly would refuse o compile this